pyfa_tool/modules/describe_module.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Dec 13 12:22:26 2023
+
+@author: thoverga
+"""
+
+import sys
+from datetime import datetime, timedelta
+from .IO import read_json
+
+
+def _split_fields(fieldslist):
+    """Split the 2d fields from the 3d fields (other representation in describe)."""
+    single_lvl_fields = []
+    multi_lvl_fields = {}
+
+    for field in fieldslist:
+        fieldname = field['name'].strip()
+
+        if (fieldname.startswith('S') & fieldname[1:3].isnumeric()):
+            # For 3d fields
+            basis_fieldname = fieldname[4:]
+            if not basis_fieldname in multi_lvl_fields.keys():
+                init_fielddict = field
+                init_fielddict['full_name'] = fieldname
+                init_fielddict['name'] = basis_fieldname
+                init_fielddict['levels'] = [int(fieldname[1:4])]
+
+                multi_lvl_fields[basis_fieldname] = init_fielddict #update dict
+
+            else:
+                # just add level to knonw levels
+                multi_lvl_fields[basis_fieldname]['levels'].append(int(fieldname[1:4]))
+        else:
+            # For 2d fields
+            single_lvl_fields.append(field)
+
+    return multi_lvl_fields, single_lvl_fields
+
+
+def _print_fields_table(fields_2d, fields_3d):
+    """ Print out the fields information."""
+    # First 2d fields
+    print('########## 2D ######### \n')
+    print(f'{"name".ljust(19)}{"index".ljust(8)}{"spectral".ljust(10)}{"nbits".ljust(6)}')
+    print('------------------------------------------')
+    for field in fields_2d:
+        print(_format_2d_field(field))
+
+    print('\n########## 3D ######### \n')
+    print(f'{"Base-name".ljust(19)}{"Full-name example".ljust(19)}{"spectral".ljust(10)}{"nbits".ljust(6)}levels')
+    print('-------------------------------------------------------------')
+    for field in fields_3d.values():
+        print(_format_3d_field(field))
+
+
+def describe_fa_from_json(metadata, fieldslist):
```

Comment:
It looks very good. Not all fields are shown? Is this done on purpose? Or are they grouped in the 3D Fields?

```bash
#################
##   VALIDITY  ##
#################

Validity               : 2011-01-13 00:00:00
Basedate               : 2011-01-01 00:00:00
Leadtime               : 12 days, 0:00:00

Timestep               : 0:05:00
N time iterations      : 3456


#######################
## HORIZONTAL GEOMETRY
#######################

Points of C+I in X      : 499
Points of C+I in Y      : 499

Kind of projection      : lcc
Reference lat (ELAT0)   : 49
Reference lon (ELON0)   : 9.9
Center lat              : 49
Center lon              : 9.9

Resolution in X (m)     : 12500
Resolution in Y (m)     : 12500


#######################
## Vertical GEOMETRY
#######################
N levels                : 46
Ref pressure            : 101325

Vert coords A           :
[0, 0.0001, 0.0003, 0.0007, 0.0013, 0.0023, 0.0045, 0.0051, 0.0073, 0.0092, 0.0108, 0.0123, 0.0135, 0.0145, 0.0154, 0.016, 0.0165, 0.0168, 0.0169, 0.0169, 0.0168, 0.0165, 0.0161, 0.0157, 0.0151, 0.0144, 0.0137, 0.0129, 0.0121, 0.0112, 0.0103, 0.0094, 0.0085, 0.0075, 0.0066, 0.0057, 0.0048, 0.004, 0.0032, 0.0025, 0.0018, 0.0012, 0.0008, 0.0004, 0.0001, 0, 0]

Vert coords B           :
[0, 0, 0, 0, 0, 0, 0, 0.0035, 0.0102, 0.0201, 0.0328, 0.0484, 0.0666, 0.0871, 0.11, 0.1349, 0.1617, 0.1903, 0.2204, 0.2519, 0.2846, 0.3184, 0.3531, 0.3884, 0.4243, 0.4606, 0.497, 0.5335, 0.5698, 0.6057, 0.6412, 0.676, 0.7099, 0.7429, 0.7746, 0.805, 0.8338, 0.8609, 0.8862, 0.9094, 0.9304, 0.949, 0.965, 0.9783, 0.9887, 0.996, 1]



#######################
## FIELDS
#######################

Number of fields        : [410]

    
########## 2D ######### 

name               index   spectral  nbits 
------------------------------------------
PROFTEMPERATURE     8       False   16.0  
PROFRESERV.EAU      9       False   16.0  
PROFRESERV.GLACE    10      False   16.0  
SURFRESERV.NEIGE    11      False   16.0  
SURFALBEDO NEIGE    12      False   16.0  
SURFDENSIT.NEIGE    13      False   16.0  
SURFALBEDO HISTO    14      False   16.0  
SURFTEMPERATURE     15      False   16.0  
SURFRESERV.EAU      16      False   16.0  
SURFRESERV.INTER    17      False   16.0  
SURFRESERV.GLACE    18      False   16.0  
SURFZ0.FOIS.G       19      False   nan   
SURFALBEDO          20      False   16.0  
SURFEMISSIVITE      21      False   16.0  
SURFET.GEOPOTENT    22      False   16.0  
SURFIND.TERREMER    23      False   16.0  
SURFPROP.VEGETAT    24      False   16.0  
SURFVAR.GEOP.ANI    25      False   16.0  
SURFVAR.GEOP.DIR    26      False   16.0  
SURFPROP.ARGILE     27      False   16.0  
SURFPROP.SABLE      28      False   16.0  
SURFEPAIS.SOL       29      False   16.0  
SURFIND.VEG.DOMI    30      False   16.0  
SURFRESI.STO.MIN    31      False   16.0  
SURFIND.FOLIAIRE    32      False   16.0  
SURFRES.EVAPOTRA    33      False   16.0  
SURFGZ0.THERM       34      False   nan   
SURFALBEDO.SOLNU    35      False   16.0  
SURFALBEDO.VEG      36      False   16.0  
SURFAEROS.SEA       37      False   16.0  
SURFAEROS.LAND      38      False   16.0  
SURFAEROS.SOOT      39      False   16.0  
SURFAEROS.DESERT    40      False   16.0  
SURFA.OF.OZONE      41      False   16.0  
SURFB.OF.OZONE      42      False   16.0  
SURFC.OF.OZONE      43      False   16.0  
SUNSHI. DURATION    44      False   16.0  
SURFNEBUL.TOTALE    183     False   16.0  
SURFNEBUL.CONVEC    184     False   16.0  
SURFNEBUL.HAUTE     185     False   16.0  
SURFNEBUL.MOYENN    186     False   16.0  
SURFNEBUL.BASSE     187     False   16.0  
SURFCAPE.MOD.XFU    188     False   16.0  
CLPMOCON.MOD.XFU    189     False   16.0  
CLSVENT.ZONAL       190     False   16.0  
CLSVENT.MERIDIEN    191     False   16.0  
CLSTEMPERATURE      192     False   16.0  
CLSHUMI.RELATIVE    193     False   16.0  
CLSHUMI.SPECIFIQ    194     False   16.0  
CLSMINI.H.TEMPER    195     False   16.0  
CLSMAXI.H.TEMPER    196     False   16.0  
CLPMHAUT.MOD.XFU    197     False   16.0  
CLSU.RAF.MOD.XFU    198     False   16.0  
CLSV.RAF.MOD.XFU    199     False   16.0  
SURFTENS.DMOG.ZO    200     False   16.0  
SURFTENS.DMOG.ME    201     False   16.0  
SURFTENS.TURB.ZO    202     False   16.0  
SURFTENS.TURB.ME    203     False   16.0  
SURFCFU.Q.TURBUL    204     False   16.0  
SURFCFU.CT.TURBU    205     False   16.0  
SURFPREC.EAU.CON    206     False   16.0  
SURFFL.COND.L.CO    207     False   16.0  
SURFPREC.NEI.CON    208     False   16.0  
SURFFL.COND.N.CO    209     False   16.0  
SURFPREC.EAU.GEC    210     False   16.0  
SURFFL.COND.L.GE    211     False   16.0  
SURFPREC.NEI.GEC    212     False   16.0  
SURFFL.COND.N.GE    213     False   16.0  
SOMMFLU.RAY.SOLA    214     False   16.0  
SURFFLU.RAY.SOLA    215     False   16.0  
SOMMFLU.RAY.THER    216     False   16.0  
SURFFLU.RAY.THER    217     False   16.0  
SURFFLU.LAT.MEVA    218     False   16.0  
SURFFLU.LAT.MSUB    219     False   16.0  
SURFFLU.MEVAP.EA    220     False   16.0  
SURFFLU.MSUBL.NE    221     False   16.0  
SURFFLU.CHA.SENS    222     False   16.0  
SURFRAYT DIR SUR    227     False   16.0  
TOPRAYT DIR SOM     228     False   16.0  
SURFDIR NORM IRR    229     False   16.0  
SURFRAYT SOLA DE    230     False   16.0  
SURFRAYT THER DE    231     False   16.0  
SURFPRESSION        372     True    18.0  
SPECSURFGEOPOTEN    373     True    nan   

########## 3D ######### 

Base-name          Full-name example  spectral  nbits levels
-------------------------------------------------------------
SNOW                S001SNOW           False     16.0  1 - 46
RAIN                S001RAIN           False     16.0  1 - 46
RK_QCTEND           S001RK_QCTEND      False     16.0  1 - 46
RAYT SOL CL         S001RAYT SOL CL    False     16.0  [1, 46]
RAYT THER CL        S001RAYT THER CL   False     16.0  [1, 46]
HUMI.SPECIFI        S001HUMI.SPECIFI   True      18.0  1 - 46
TEMPERATURE         S001TEMPERATURE    True      18.0  1 - 46
WIND.U.PHYS         S001WIND.U.PHYS    True      18.0  1 - 46
WIND.V.PHYS         S001WIND.V.PHYS    True      18.0  1 - 46

```

410 fields but not all printed.

Reply:
This is done one purpose. The fields at the bottom are 3D fields, and consists of 2d fields at specific levels (see 'levels' column in the printout. 

In your example, this means that the fields S001HUMI.SPECIFI, S002HUMI.SPECIFI, ... are not printed out since they are interpreted as a single 3D field.